### ISSUE TITLE
Are you aware of Tyxml ?

### ISSUE THREAD

**Issue by Drup** [rbardou/funweb#1]
See the [Tyxml_js](http://ocsigen.org/js_of_ocaml/2.7/api/Tyxml_js) module and the [Tyxml manual](https://ocsigen.org/tyxml/dev/manual/). We also recently introduced a [ppx](https://ocsigen.org/tyxml/dev/manual/ppx) that allows to write HTML/SVG and can be interpreted, among other things, in js_of_ocaml.

**Comment by doomeer** [comment]
Thanks for taking a look!

A long time ago I used some part of Ocsigen to write HTML trees but I found it rather annoying to use in practice. I don't remember how it was called, maybe it is now Tyxml. Thanks for the links.

To be honest though, writing HTML trees is probably the least interesting issue that Funweb tries to solve. I think the key feature of Funweb is rather that it adopts ideas from React (see http://jlongster.com/Removing-User-Interface-Complexity,-or-Why-React-is-Awesome). Basically, the idea is to rebuild the DOM tree everytime one wants to modify it. This solves many issues but it also creates new ones. I tried to solve those new issues with very simple solutions and I came up with Funweb.

Others have tackled this problem, Funweb is not ground-breaking, but it was a nice challenge and I had fun designing it :D

**Comment by rbardou** [comment]
Thanks for taking a look!

A long time ago I used some part of Ocsigen to write HTML trees but I found it rather annoying to use in practice. I don't remember how it was called, maybe it is now Tyxml. Thanks for the links.

To be honest though, writing HTML trees is probably the least interesting issue that Funweb tries to solve. I think the key feature of Funweb is rather that it adopts ideas from React (see http://jlongster.com/Removing-User-Interface-Complexity,-or-Why-React-is-Awesome). Basically, the idea is to rebuild the DOM tree everytime one wants to modify it. This solves many issues but it also creates new ones. I tried to solve those new issues with very simple solutions and I came up with Funweb.

Others have tackled this problem, Funweb is not ground-breaking, but it was a nice challenge and I had fun designing it :D

**Comment by Drup** [comment]
@rbardou Indeed, this is also covered by the [`Tyxml_js.R` module](http://ocsigen.org/js_of_ocaml/2.7/api/Tyxml_js.R). There is a simple example [here](https://github.com/talex5/js-skeleton) and a [manual](http://ocsigen.org/eliom/5.0/manual/clientserver-html#reactive) in eliom.

**Comment by Drup** [comment]
(And I agree, reactive UIs for the web are great)
